src/main/java/com/faforever/client/chat/PircBotXChatService.java

```diff
@@ -426,4 +446,42 @@ public void onModeratorSet(String channelName, String username) {
     }
     chatUser.getModeratorInChannels().add(channelName);
   }
+
+  @Override
+  public Collection<Color> getAssignedColors() {
+    return assignedColors;
+  }
+
+  @Override
+  public ImmutableSortedSet<Channel> getChannelsForUser(String username) {
+    return pircBotX.getUserChannelDao().getChannels(pircBotX.getUserChannelDao().getUser(username));
+  }
+
+  @Override
+  public ChatUser getChatUser(String username) {
+    for (String channel : chatUserLists.keySet()) {
+      if (chatUserLists.get(channel).containsKey(username)) {
```

Comment:
and store `chatUserLists.get(channel)` to variable (that is `entry.getValue()` when iterating over entry set)